agent/asm/asm.go

```diff
@@ -87,6 +89,34 @@ func extractASMValue(out *secretsmanager.GetSecretValueOutput) (types.AuthConfig
 	return dac, nil
 }
 
+
+func GetSecretFromASMWithInput(input *secretsmanager.GetSecretValueInput,
+	client secretsmanageriface.SecretsManagerAPI, jsonKey string) (string, error) {
+	out, err := client.GetSecretValue(input)
+	if err != nil {
+		return "", errors.Wrapf(err, "secret %s", *input.SecretId)
```

Comment:
based on this line in asmsecret.go:
```
input.SecretId = pointerOrNil(reconstructASMARN(arnObj))
```
seems like input.SecretId might be nil here (and will panic when you deference it)? i think the caller should return an error before calling GetSecretFromASMWithInput if input.SecretId is nil

Reply:
good point, checked for secret id before calling GetSecretFromASMWithInput